sdk/src/store.rs

```diff
@@ -2712,4 +2818,158 @@ pub mod tests {
+
+        // set claim for side car generation
+        claim.set_remote_manifest(RemoteManifest::SideCar).unwrap();
+
+        // Do we generate JUMBF?
+        let signer = temp_signer();
+
+        store.commit_claim(claim).unwrap();
+
+        let saved_manifest = store.save_to_asset(&ap, &signer, &op).unwrap();
+
+        assert!(sidecar.exists());
+
+        // load external manifest
+        let loaded_manifest = std::fs::read(sidecar).unwrap();
+
+        // compare returned to external
+        assert_eq!(saved_manifest, loaded_manifest);
+
+        // load the jumbf back into a store
+        let mut validation_log = OneShotStatusTracker::default();
+        let restored_store = Store::from_jumbf(&loaded_manifest, &mut validation_log).unwrap();
+        let pc = restored_store.provenance_claim().unwrap();
+
+        // make sure this manifest goes with this asset
+        for dh_assertion in pc.data_hash_assertions() {
+            if dh_assertion.label_root() == DataHash::LABEL {
+                let dh = DataHash::from_assertion(dh_assertion).unwrap();
+
+                dh.verify_hash(&op.clone(), Some(pc.alg().to_string()))
+                    .unwrap();
+            }
+        }
+    }
+
+    #[test]
+    fn test_external_manifest_embedded() {
+        // test adding to actual image
+        let ap = fixture_path("libpng-test.png");
+        let temp_dir = tempdir().expect("temp dir");
+        let op = temp_dir_path(&temp_dir, "libpng-test-c2pa.png");
+
+        let sidecar = op.with_extension(MANIFEST_STORE_EXT);
+
+        // Create claims store.
+        let mut store = Store::new();
+
+        // Create a new claim.
+        let mut claim = create_test_claim().unwrap();
+
+        // Do we generate JUMBF?
+        let signer = temp_signer();
+
+        // start with base url
+        let fp = format!("file:/{}", sidecar.to_str().unwrap());
+        let url = url::Url::parse(&fp).unwrap();
+
+        let url_string: String = url.into();
+
+        // set claim for side car with remote manifest embedding generation
+        claim
+            .set_remote_manifest(RemoteManifest::Remote(url_string.clone()))
+            .unwrap();
+
+        store.commit_claim(claim).unwrap();
+
+        let saved_manifest = store.save_to_asset(&ap, &signer, &op).unwrap();
+
+        assert!(sidecar.exists());
+
+        // load external manifest
+        let loaded_manifest = std::fs::read(sidecar).unwrap();
+
+        // compare returned to external
+        assert_eq!(saved_manifest, loaded_manifest);
+
+        // load the jumbf back into a store
+        let mut validation_log = OneShotStatusTracker::default();
+        let restored_store = Store::from_jumbf(&loaded_manifest, &mut validation_log).unwrap();
+        let pc = restored_store.provenance_claim().unwrap();
+
+        let mut asset_reader = std::fs::File::open(op.clone()).unwrap();
+        let handler = get_cailoader_handler("png").unwrap();
```

Comment:
You can use:  
let ext_ref = XmpInfo::from_source(asset_reader, "png").provenance.unwrap();